engine/src/graphics/vkHelpers.cpp

```diff
@@ -23,136 +23,112 @@ VULKAN_HPP_DEFAULT_DISPATCH_LOADER_DYNAMIC_STORAGE
 namespace vk::util
 {
 	// Callback function used for Vulkan debug messaging.
-	VkBool32 debugUtilsMessengerCallback(VkDebugUtilsMessageSeverityFlagBitsEXT messageSeverity,
-										 VkDebugUtilsMessageTypeFlagsEXT messageTypes,
-										 const VkDebugUtilsMessengerCallbackDataEXT * pCallbackData,
-										 void * /*pUserData*/)
+	VkBool32 debugUtilsMessengerCallback(VkDebugUtilsMessageSeverityFlagBitsEXT messageSeverity, VkDebugUtilsMessageTypeFlagsEXT messageTypes,
+										 const VkDebugUtilsMessengerCallbackDataEXT * pCallbackData, void * /* pUserData */)
 	{
 #ifndef GEN_NDEBUG
 		// In debug mode, certain Vulkan validation layer messages are ignored.
 		// Here, we filter out specific validation messages identified by their messageIdNumber.
-		if ( pCallbackData->messageIdNumber == 648835635 ) //NOLINT UNASSIGNED-khronos-Validation-debug-build-warning-message
+		if (pCallbackData->messageIdNumber == 648835635) // NOLINT UNASSIGNED-khronos-Validation-debug-build-warning-message
 		{
 			return vk::False;
 		}
-		if ( pCallbackData->messageIdNumber == 767975156 ) //NOLINT UNASSIGNED-BestPractices-vkCreateInstance-specialuse-extension
+		if (pCallbackData->messageIdNumber == 767975156) // NOLINT UNASSIGNED-BestPractices-vkCreateInstance-specialuse-extension
 		{
 			return vk::False;
 		}
 #endif
 
-		// Log the Vulkan debug message.
-		gen::logger::error("vulkan", std::format("{}: {}:\n "
-												 "\tmessageIDName   = <{}>\n"
-												 "\tmessageIDNumber = {}\n"
-												 "\tmessage         = <{}>\n",
-												 vk::to_string(static_cast<vk::DebugUtilsMessageSeverityFlagBitsEXT>( messageSeverity ) ),
-												 vk::to_string( static_cast<vk::DebugUtilsMessageTypeFlagsEXT>( messageTypes ) ),
-												 pCallbackData->pMessageIdName,
-												 pCallbackData->messageIdNumber,
-												 pCallbackData->pMessage));
+		if (messageSeverity == VkDebugUtilsMessageSeverityFlagBitsEXT::VK_DEBUG_UTILS_MESSAGE_SEVERITY_ERROR_BIT_EXT)
+		{
+			// Any actual errors from vulkan will be logged with an error severity
+			gen::logger::error("vulkan", std::format("{}: {}:\n "
+													 "\tmessageIDName   = <{}>\n"
+													 "\tmessageIDNumber = {}\n"
+													 "\tmessage         = <{}>\n",
+													 vk::to_string(static_cast<vk::DebugUtilsMessageSeverityFlagBitsEXT>(messageSeverity)),
+													 vk::to_string(static_cast<vk::DebugUtilsMessageTypeFlagsEXT>(messageTypes)), pCallbackData->pMessageIdName,
+													 pCallbackData->messageIdNumber, pCallbackData->pMessage));
+		}
+		else
+		{
+			// Everything else will be treated as a warning.
```

Comment:
The function that creates the debug messenger registers for levels beyond just errors and warnings; eg we'll get a notification that the swapchain is out of date when the window is resized, I don't think we want to treat that as a warning.

Reply:
Then how about we treat only vk errors and warnings as a error through the logger and treat everything else as a warning. Does that sound reasonable or is there something else I’ve not considered?